generic-examples/kafka/README.md

```diff
@@ -1,30 +1,52 @@
+
+You can find more information about Apache Camel and Apache Camel K on the [official Camel website](https://camel.apache.org).
+
+## Before you begin
+
+Read the general instructions in the [root README.md file](/README.md) for setting up your environment and the Kubernetes cluster before looking at this example.
+
+Make sure you've read the [installation instructions](https://camel.apache.org/camel-k/latest/installation/installation.html) for your specific
+cluster before starting the example.
+
+## Additional Requirements for running the examples
+**A Kafka broker**: handles the storage and passing of messages.
 
 ## Authentication to Kafka
 
-We have a dedicated section to show [how to authenticate Camel K to Kafka](./sasl_ssl/).
+To use Kafka with authentication, we have a dedicated section to show [how to authenticate Camel K to Kafka](./sasl_ssl/). \
+For a simple use case without client authentication, continue with this guide.
+
+## Understanding the example
+- [SampleKafkaConsumer.java](./SampleKafkaConsumer.java): contains a route that reads message from a kafka topic and logs the message
+- [application.properties](./application.properties): holds properties required to connect to kafka broker and read from topic.
 
 ## Simple usage
 
-To run this example first set-up Kafka on your k8s cluster.
-A convenient way to do so is by using the Strimzi project, if you are using minikube follow these instructions at https://strimzi.io/quickstarts/minikube/
+To run this example, first set-up Kafka on your k8s cluster.
+A convenient way to do so is by using the Strimzi project. Visit https://strimzi.io/quickstarts/ for set-up instructions. For the instructions on the linked site, it will suffice to only apply the Strimzi installation file and provision the kafka cluster.
 
-For convenience create a configmap to contain the properties:
+IMPORTANT: The `kafka.host` value in `application.properties` needs to be set to the CLUSTER-IP address of the my-cluster-kafka-bootstrap service in the kafka namespace. To do this run:
 ```
-kubectl create configmap kafka.props  --from-file=examples/kafka/application.properties
+kafkaip=`kubectl get svc/my-cluster-kafka-bootstrap -n kafka -ojsonpath="{.spec.clusterIP}"`; sed -i "/kafka\.host/s/<.*>/$kafkaip/g" application.properties
+```
+
+Create a configmap to contain the properties:
+```
+kubectl create configmap kafka.props  --from-file=application.properties
 ```
 
-IMPORTANT: The kafka.host value in application.properties needs to be set to the CLUSTER-IP address of the my-cluster-kafka-bootstrap service in the kafka namespace:
- `kubectl get services -n kafka | grep my-cluster-kafka-bootstrap | awk '/[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/ {print $3}'`
+
 
 Finally run this sample using the command:
 ```
-./kamel run examples/kafka/SampleKafkaConsumer.java --config=configmap:kafka.props
+kamel run SampleKafkaConsumer.java --config=configmap:kafka.props --dev
```

Comment:
Here we don't specifically need `--dev`.

Reply:
I only added it because the consumer logs the messages but I can see how it's unnecessary. Done